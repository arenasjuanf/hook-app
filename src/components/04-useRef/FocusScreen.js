import React, { useRef } from 'react';
import './focus.css';

export const FocusScreen = () => {

    const inputRef = useRef();
    
    const handleClick = () => {
        inputRef.current.select();
    }

    return (
        <div>
            <h1>Focus Input</h1>
            <hr />
            <form className="form-group">
                <input 
                    ref={ inputRef }
                    className="form-control"
                    placeholder="Your Name"
                />
                <div className="text-end">
                    <button 
                        type="button"
                        className="btn btn-outline-primary mt-1"
                        onClick={ handleClick }
                    >
                        Focus
                    </button>
                </div>
            </form>
        </div>
    )
}
