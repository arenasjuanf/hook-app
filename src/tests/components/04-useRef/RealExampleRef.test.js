import { shallow } from "enzyme"
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef"

describe("<RealExampleRef /> tests", () => {
    const wrapper = shallow(<RealExampleRef />);
    
    test("it should show successfully", () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    })

    test("it should show <MultipleCustomHooks /> successfully", () => {

        wrapper.find("button").simulate("click");
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    })
})