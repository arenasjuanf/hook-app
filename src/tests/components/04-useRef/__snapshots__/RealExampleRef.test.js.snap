// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RealExampleRef /> tests it should show successfully 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show
  </button>
</div>
`;
